clients/branch.go

```diff
@@ -28,13 +28,13 @@ type BranchProtectionRule struct {
 	AllowForcePushes           *bool
 	RequireLinearHistory       *bool
 	EnforceAdmins              *bool
-	RequiredStatusChecks       StatusChecksRule
+	RequiredStatusChecks       *StatusChecksRule
 }
 
 // StatusChecksRule captures settings on status checks.
 type StatusChecksRule struct {
-	Strict   *bool
 	Contexts []string
+	UpToDate bool
```

Comment:
Could you explain the reasoning behind renaming this? This `bool` still captures `Strict` correct?

Reply:
took me 10mn to understand what `strict` meant in the GitHub UX settings. I can't find it in the API either. It may have been the old name they gave, I don't know. So I simply renamed it.. maybe `UpToDateBeforeMerge` would be more accurate. Let me know if you prefer `Strict` or if you're fine renaming it to something more meaningful.

Reply:
I'm ok with the rename, was just curious to understand the reasoning.

Nit: if `Strict` is confusing, how about `Enforced`?

Reply:
enforced is confusing: enforcement of what? It could replace RequiresStatusChecks but not Strict, I think.
Note that it's even more complicated than that: in the UX, GH has a warning that `This setting will not take effect unless at least one status check is enabled`. We will check that in the check itself, though.

Reply:
Ack.